fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt

```diff
@@ -129,6 +88,95 @@ fun ProductAnalysisError(
     }
 }
 
+@Suppress("LongMethod")
```

Comment:
had to suppress longMethod, maybe the ReviewQualityCheckInfoType could be set to Info to start?

Reply:
I see, let's keep this when logic in the above function in that case. 

We don't want to set a var and then overwrite in, it's better to be declarative and have one state.

I think Triple has fulfilled it's usecase here, let's create a data class with 5 properties that are required for the `InfoCard` composable. Then each error state maps to that object which is a param for `ReviewQualityCheckInfoCard`.

Reply:
I moved the `when` logic back into the above function, but didn't add the data class, since we would still have to check if it's `Error.ProductNotAvailable` to create a ReviewQualityCheckInfoCard with button. It only seemed marginally better. But if you have something else in mind, maybe I can add as a follow up, if we're trying to get this in before code freeze.

Reply:
That's fair, we can discuss that approach after landing this 👍